diff -r a3d508507022 gfx/cairo/cairo/src/cairo-compiler-private.h
--- a/gfx/cairo/cairo/src/cairo-compiler-private.h	Mon Feb 24 15:57:17 2025 +0000
+++ b/gfx/cairo/cairo/src/cairo-compiler-private.h	Mon Feb 24 23:54:13 2025 -0500
@@ -144,7 +144,7 @@
 #define unlikely(expr) (expr)
 #endif
 
-#ifndef __GNUC__
+#if !defined(__GNUC__) && !defined (__clang__)
 #undef __attribute__
 #define __attribute__(x)
 #endif
