Index: mozilla/security/nss/cmd/addbuiltin/addbuiltin.c
===================================================================
RCS file: /cvsroot/mozilla/security/nss/cmd/addbuiltin/addbuiltin.c,v
retrieving revision 1.19
diff -u -8 -p -r1.19 addbuiltin.c
--- mozilla/security/nss/cmd/addbuiltin/addbuiltin.c	4 Jul 2012 15:21:47 -0000	1.19
+++ mozilla/security/nss/cmd/addbuiltin/addbuiltin.c	27 Nov 2012 03:48:11 -0000
@@ -468,16 +468,29 @@ int main(int argc, char **argv)
 	    "%s: Cannot change stdin to binary mode. Use -i option instead.\n",
 	            progName);
 	    exit(1);
 	}
 #endif
 	infile = PR_STDIN;
     }
 
+#if defined(WIN32)
+    /* We must put stdout into O_BINARY mode or else the output will include
+    ** carriage returns.
+    */
+    {
+	int smrv = _setmode(_fileno(stdout), _O_BINARY);
+	if (smrv == -1) {
+	    fprintf(stderr, "%s: Cannot change stdout to binary mode.\n", progName);
+	    exit(1);
+	}
+    }
+#endif
+
     nickname = strdup(addbuiltin.options[opt_Nickname].arg);
     
     NSS_NoDB_Init(NULL);
 
     if (addbuiltin.options[opt_Distrust].activated ||
         addbuiltin.options[opt_DistrustCRL].activated) {
       addbuiltin.options[opt_ExcludeCert].activated = PR_TRUE;
       addbuiltin.options[opt_ExcludeHash].activated = PR_TRUE;
