diff --git a/sqlite-vec.c b/sqlite-vec.c
index 86378ba..e8e38c9 100644
--- a/sqlite-vec.c
+++ b/sqlite-vec.c
@@ -61,18 +61,6 @@ SQLITE_EXTENSION_INIT3
 #define LONGDOUBLE_TYPE long double
 #endif
 
-#ifndef _WIN32
-#ifndef __EMSCRIPTEN__
-#ifndef __COSMOPOLITAN__
-#ifndef __wasi__
-typedef u_int8_t uint8_t;
-typedef u_int16_t uint16_t;
-typedef u_int64_t uint64_t;
-#endif
-#endif
-#endif
-#endif
-
 typedef int8_t i8;
 typedef uint8_t u8;
 typedef int16_t i16;
