From acfded8ef5cabfdc868583b6c5e5b89dfff2b466 Mon Sep 17 00:00:00 2001
From: serge-sans-paille <sguelton@mozilla.com>
Date: Tue, 13 May 2025 12:16:53 +0200
Subject: [PATCH 4/5] Disable ORT Stream

---
 cmake/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 98cb298af0..1022626b8d 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -1114,7 +1114,7 @@ function(onnxruntime_set_compile_flags target_name)
     endif()
 	# Enable stream for all the non-minimal build
     if (NOT onnxruntime_MINIMAL_BUILD)
-      target_compile_definitions(${target_name} PRIVATE ORT_ENABLE_STREAM)
+      #target_compile_definitions(${target_name} PRIVATE ORT_ENABLE_STREAM)
     endif()
 	if (onnxruntime_REDUCED_OPS_BUILD)
       target_compile_definitions(${target_name} PRIVATE REDUCED_OPS_BUILD)
-- 
2.49.0

