diff --git a/mingw-w64-crt/lib32/wintrust.def b/mingw-w64-crt/lib32/wintrust.def
index fbff4e75d..8ed58036b 100644
--- a/mingw-w64-crt/lib32/wintrust.def
+++ b/mingw-w64-crt/lib32/wintrust.def
@@ -18,8 +18,10 @@ WTHelperCertFindIssuerCertificate@28
 AddPersonalTrustDBPages@12
 CatalogCompactHashDatabase@16
 CryptCATAdminAcquireContext@12
+CryptCATAdminAcquireContext2@20
 CryptCATAdminAddCatalog@16
 CryptCATAdminCalcHashFromFileHandle@16
+CryptCATAdminCalcHashFromFileHandle2@20
 CryptCATAdminEnumCatalogFromHash@20
 CryptCATAdminPauseServiceForBackup@8
 CryptCATAdminReleaseCatalogContext@12
diff --git a/mingw-w64-crt/lib64/wintrust.def b/mingw-w64-crt/lib64/wintrust.def
index c2d68ffee..9df52ad22 100644
--- a/mingw-w64-crt/lib64/wintrust.def
+++ b/mingw-w64-crt/lib64/wintrust.def
@@ -19,8 +19,10 @@ WTHelperCertFindIssuerCertificate
 AddPersonalTrustDBPages
 CatalogCompactHashDatabase
 CryptCATAdminAcquireContext
+CryptCATAdminAcquireContext2
 CryptCATAdminAddCatalog
 CryptCATAdminCalcHashFromFileHandle
+CryptCATAdminCalcHashFromFileHandle2
 CryptCATAdminEnumCatalogFromHash
 CryptCATAdminPauseServiceForBackup
 CryptCATAdminReleaseCatalogContext
