diff --git a/expat/lib/xmlparse.c b/expat/lib/xmlparse.c
--- a/expat/lib/xmlparse.c	2024-09-12 17:10:00
+++ b/expat/lib/xmlparse.c	2024-09-12 17:10:21
@@ -966,6 +966,8 @@
 
 static unsigned long
 generate_hash_secret_salt(XML_Parser parser) {
+/* BEGIN MOZILLA CHANGE (we already set a salt through XML_SetHashSalt) */
+#if 0
   unsigned long entropy;
   (void)parser;
 
@@ -1005,6 +1007,10 @@
                          entropy * (unsigned long)2305843009213693951ULL);
   }
 #endif
+#else
+  abort();
+#endif
+/* END MOZILLA CHANGE */
 }
 
 static unsigned long
