diff --git a/expat/lib/xmlparse.c b/expat/lib/xmlparse.c
--- a/expat/lib/xmlparse.c
+++ b/expat/lib/xmlparse.c
@@ -960,6 +960,8 @@ ENTROPY_DEBUG(const char *label, unsigne
 
 static unsigned long
 generate_hash_secret_salt(XML_Parser parser) {
+/* BEGIN MOZILLA CHANGE (we already set a salt through XML_SetHashSalt) */
+#if 0
   unsigned long entropy;
   (void)parser;
 
@@ -999,6 +1001,10 @@ generate_hash_secret_salt(XML_Parser par
                          entropy * (unsigned long)2305843009213693951ULL);
   }
 #endif
+#else
+  abort();
+#endif
+/* END MOZILLA CHANGE */
 }
 
 static unsigned long
