diff --git a/expat/lib/xmlparse.c b/expat/lib/xmlparse.c
--- a/expat/lib/xmlparse.c	2024-09-12 17:10:49
+++ b/expat/lib/xmlparse.c	2024-09-12 17:11:34
@@ -2324,6 +2324,10 @@
   XmlUpdatePosition(parser->m_encoding, parser->m_positionPtr,
                     parser->m_bufferPtr, &parser->m_position);
   parser->m_positionPtr = parser->m_bufferPtr;
+/* BEGIN MOZILLA CHANGE (always set m_eventPtr/m_eventEndPtr) */
+  parser->m_eventPtr = parser->m_bufferPtr;
+  parser->m_eventEndPtr = parser->m_bufferPtr;
+/* END MOZILLA CHANGE */
   return result;
 }
 
