diff --git a/expat/lib/xmlparse.c b/expat/lib/xmlparse.c
--- a/expat/lib/xmlparse.c	2024-09-12 17:23:02
+++ b/expat/lib/xmlparse.c	2024-09-12 17:23:42
@@ -3013,9 +3013,15 @@
           if (parser->m_skippedEntityHandler)
             parser->m_skippedEntityHandler(parser->m_handlerArg, entity->name,
                                            0);
+/* BEGIN MOZILLA CHANGE (Bug 35984 - Undeclared entities are ignored when external DTD not found) */
+#if 0
           else if (parser->m_defaultHandler)
             reportDefault(parser, enc, s, next);
           break;
+#else
+          return XML_ERROR_UNDEFINED_ENTITY;
+#endif
+/* END MOZILLA CHANGE */
         }
         result = processInternalEntity(parser, entity, XML_FALSE);
         if (result != XML_ERROR_NONE)
@@ -6196,7 +6202,13 @@
         if ((pool == &parser->m_tempPool) && parser->m_defaultHandler)
           reportDefault(parser, enc, ptr, next);
         */
+/* BEGIN MOZILLA CHANGE (Bug 35984 - Undeclared entities are ignored when external DTD not found) */
+#if 0
         break;
+#else
+        return XML_ERROR_UNDEFINED_ENTITY;
+#endif
+/* END MOZILLA CHANGE */
       }
       if (entity->open) {
         if (enc == parser->m_encoding) {
