diff --git a/intl/icu_capi/src/data_struct.rs b/intl/icu_capi/src/data_struct.rs
--- a/intl/icu_capi/src/data_struct.rs
+++ b/intl/icu_capi/src/data_struct.rs
@@ -21,7 +21,7 @@ pub mod ffi {
     ///
     /// This can be used to construct a StructDataProvider.
     #[diplomat::attr(*, disable)]
-    pub struct ICU4XDataStruct(pub(crate) AnyPayload);
+    pub struct ICU4XDataStruct(#[allow(dead_code)] AnyPayload);
 
     impl ICU4XDataStruct {
         /// Construct a new DecimalSymbolsV1 data struct.
